### PR TITLE
Removing auth from redux

### PR DESCRIPTION
**Pull request by SamGodwin2**
### What is the context of this PR?
Removed auth from redux.

### How to review 
New tests should pass and the app should behave in the same way as before.

### CODE REVIEW

File: eq-author/src/App/AccessDeniedPage/index.test.js [hunk 1]
```diff
@@ -1,11 +1,23 @@
 import React from "react";
 import { render } from "tests/utils/rtl";
 
+import { MeContext } from "App/MeContext";
+
 import AccessDenied from "./";
 
 describe("Access denied page", () => {
+  let me;
```
[the remark this answers is not in the document]
Comment (SamGodwin2) [reply eq-author/src/App/AccessDeniedPage/index.test.js]: Think I originally had another test here that didn't end up being needed, I'll move it down into the test.

File: eq-author/src/App/MeContext.test.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    });
+
+    afterEach(() => {
+      localStorage.clear();
+      window.fetch = originalFetch;
+    });
+
+    it("Should sign the user in.", () => {
+      Component = props => (
+        // eslint-disable-next-line
```
[the remark this answers is not in the document]
Comment (SamGodwin2) [reply eq-author/src/App/MeContext.test.js]: It's not happy about my onClick function not beginning with handle and not giving the button any proptypes, I'll add the rules I'm disabling.

File: eq-author/src/apollo/createApolloErrorLink.js [hunk 3]
```diff
@@ -19,7 +18,8 @@ export const errorHandler = (getStore, error) => {
     switch (httpStatusCode) {
       // 401 - User does not exist
       case 401:
-        getStore().dispatch(signOutUser());
+        localStorage.removeItem("accessToken");
```
[the remark this answers is not in the document]
Comment (SamGodwin2) [reply eq-author/src/apollo/createApolloErrorLink.js]: I'm not sure, I tried thinking of some way of doing it, but wasn't sure how we could get the function into here as it requires the client object which isn't created until after this link is?

File: eq-author/src/redux/configureStore.js [hunk 4]
```diff
@@ -6,21 +6,18 @@ import {
...
 } from "react-router-redux";
 import saving from "redux/saving/reducer";
-import authReducer from "redux/auth/reducer";
-import auth from "components/Auth";
 
 const configureStore = (history, client, preloadedState) =>
   createStore(
     combineReducers({
       router,
       saving,
```
[the remark this answers is not in the document]
Comment (SamGodwin2) [reply eq-author/src/redux/configureStore.js]: Yup. 😀

File: eq-author/src/App/AccessDeniedPage/index.test.js [hunk 5]
```diff
@@ -1,11 +1,23 @@
...
   it("should render with access denied message", () => {
-    const { getByText } = render(<AccessDenied />);
+    me = {
+      id: "123",
+      name: "Zuko",
```
[the remark this answers is not in the document]
Comment (SamGodwin2) [reply eq-author/src/App/AccessDeniedPage/index.test.js]: I've been rumbled. 👀 😂

File: eq-author/src/apollo/createApolloErrorLink.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (SamGodwin2) [reply eq-author/src/apollo/createApolloErrorLink.js]: I ended up being kinda happy with this as it seems to have pretty much the same result and this situation is very, very unlikely to ever happen outside of dev and staging environments.

File: eq-author/src/apollo/createApolloErrorLink.test.js [hunk 6]
```diff
@@ -35,12 +35,12 @@ describe("createErrorLink", () => {
...
   });
 
-  it("should dispatch a logout event when the server responds with 401 status code", () => {
+  it("should clear the local storage if the sever responds with a 401", () => {
     errorHandler(createMockStore, {
       networkError: { statusCode: 401 },
       graphQLErrors,
     });
-    expect(dispatch).toHaveBeenCalledWith(expect.any(Function));
+    expect(localStorage.getItem("accessToken")).toBeFalsy();
```
[the remark this answers is not in the document]
Comment (SamGodwin2) [reply eq-author/src/apollo/createApolloErrorLink.test.js]: Asserted its truthy before the error is fired.

**Comment by heofs** [comment]
Just a suggestion, but could it be relevant to update the firebase dependencies in this PR? Since the change is so big I guess it would be a good opportunity to test the update properly?
<https://github.com/ONSdigital/eq-author-app/pull/424>

**Comment by tgandrews** [comment]
This is already a big change I'd rather keep it smaller than let it grow any more

File: eq-author/src/App/index.js [hunk 7]
```diff
@@ -20,55 +19,58 @@ import SignInPage from "./SignInPage";
...
+};
 
 Routes.propTypes = {
-  isSignedIn: PropTypes.bool.isRequired,
+  data: PropTypes.shape({
+    id: PropTypes.string,
+    name: PropTypes.string,
+    email: PropTypes.string,
+    picture: PropTypes.string,
+  }),
```
Comment (samiwel) [inline eq-author/src/App/index.js]: I think we can use `CustomPropTypes.user` here

File: eq-author/src/components/PrivateRoute/index.js [hunk 8]
```diff
@@ -2,27 +2,53 @@ import React from "react";
...
 PrivateRoute.propTypes = {
   component: PropTypes.func.isRequired,
-  isSignedIn: PropTypes.bool.isRequired,
+  hasAccessToken: PropTypes.bool.isRequired,
+  awaitingUserQuery: PropTypes.bool,
+  me: PropTypes.shape({
+    id: PropTypes.string,
+    name: PropTypes.string,
+    email: PropTypes.string,
+    picture: PropTypes.string,
```
Comment (samiwel) [inline eq-author/src/components/PrivateRoute/index.js]: I think we can use `CustomPropTypes.user` here

File: eq-author/src/App/MeContext.test.js [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+        refreshToken: "refreshToken",
+      };
+    });
+
+    afterEach(() => {
+      localStorage.clear();
+      window.fetch = originalFetch;
+    });
+
+    it("Should sign the user in.", () => {
```
Comment (tgandrews) [inline eq-author/src/App/MeContext.test.js]: Are we testing the error case?

File: eq-author/src/middleware/headers/authHeader.js [hunk 10]
```diff
@@ -24,17 +24,23 @@ const getQuestionnaireId = () => {
...
+
   const questionnaireId = getQuestionnaireId();
   if (questionnaireId) {
     returnedHeaders.questionnaireId = questionnaireId;
   }
 
   const decodedToken = jwt.decode(accessToken);
-  if (decodedToken.exp < Math.floor(Date.now()) / 1000) {
+
+  if (decodedToken.exp < Math.floor(Date.now()) / 1000 || !decodedToken.name) {
```
Comment (tgandrews) [inline eq-author/src/middleware/headers/authHeader.js]: Why is this change needed?

File: eq-author/src/redux/configureStore.js [hunk 11]
```diff
@@ -6,21 +6,18 @@ import {
...
-      auth: authReducer,
     }),
     preloadedState,
     composeWithDevTools(
       applyMiddleware(
         createRouterMiddleware(history),
-        thunk.withExtraArgument({ auth, client })
+        thunk.withExtraArgument({ client })
```
Comment (tgandrews) [inline eq-author/src/redux/configureStore.js]: Do we still need client?

**Comment by heofs** [comment]
Looks good, just a couple of grammar nazi comments :p

File: eq-author/src/index.js [hunk 12]
```diff
@@ -84,13 +86,15 @@ const headersLink = setContext((_, { headers }) =>
...
 const link = ApolloLink.from([
-  createErrorLink(store),
+  createErrorLink(getStore),
   headersLink,
   networkLink,
 ]);
 
 const client = createApolloClient(link, cache);
 
+store = configureStore(history, client);
```
Comment (samiwel) [inline eq-author/src/index.js]: `history` and `client` are not being used by `configureStore`, can they be removed?

File: eq-author/src/index.js [hunk 13]
```diff
@@ -84,13 +86,15 @@ const headersLink = setContext((_, { headers }) =>
 );
 
 const link = ApolloLink.from([
-  createErrorLink(store),
+  createErrorLink(getStore),
```
Comment (samiwel) [inline eq-author/src/index.js]: Is there a danger here that getStore might be called before it is initialised (below)?

Comment (samiwel) [reply eq-author/src/index.js]: Actually probably not. It's only used for sending sentry errors (which would require the app to be running)

**Comment by heofs** [comment]
Reviewed again and the changes looks good to me.

File: eq-author/src/index.js (hunk 13 above)
Comment (SamGodwin2) [reply eq-author/src/index.js]: It's one of the things I was hoping to clear up in this pr but we still need the store here for the banner in the error link. Hopefully we can get rid of this after we unpick that last bit.
